exchange/binance.go

```diff
@@ -117,11 +117,11 @@ func (b *Binance) validate(pair string, quantity float64) error {
 
 	info, ok := b.assetsInfo[pair]
 	if !ok {
-		return ErrInvalidAsset
+		return fmt.Errorf("\nPair: %s, Amount: %v\n-----\n%w", pair, quantity, ErrInvalidAsset)
```

Comment:
I think it is not a good approach.
Erros shoul not have breaklines and captalization.

Maybe, we can only include more data in Telegram module, and not in Exchange module.

Reply:
Hi @rodrigo-brito, thanks for your feedback
To be honest I had some troubles with approaching this issue.
I could do it inside telegram module however I'm not sure exactly how to exactly do it.
My only idea is to create function similar to `OnError` and pass it to the `BuyHandle`
`func (t telegram) OnBuyError(pair string, amount string, side model.SideType, err error)`.
However this approach has flaws. Because the message comes from controller: `CreateOrderMarketQuote` which uses `NotifyError` which calls the `OnError`.

I believe the change should be included in the place I proposed in my PR, however maybe there is any better idea/pattern how to implement it.

I would appreciate any tips how can I do it inside Telegram to avoid sending double notifications when we want to have more detailed info, but that's tough since not every call on `OnErrror` has to be related with buy action.

Reply:
A good approach:

Create a custom error object.
```go
type OrderError struct {
        Message string
	Side        ninjabot.OrderSide
        Size        float64
        Pair        string
        // other fields... (eg: pair, type, exchange, etc)
}

func (o *OrderError) Error() string {
	return fmt.Sprintf("order creation error: %s", o.Message)
}
```

Then, you can access the custom fields in Telegram `OnError` with this snippet:

```go
if orderErr, ok := err.(*OrderError); ok {
    // orderErr.Pair, orderErr.Side, others are available here.
}

Reply:
Reference: https://www.digitalocean.com/community/tutorials/creating-custom-errors-in-go

Reply:
Thanks for the help @rodrigo-brito , I'm doing my best however this go specific idea of error handling and type assertion is confusing me a little bit, I'm following the reference which u provided yet the `ok` value in if statement returns false, do you have any idea why?

Reply:
Okay so I actually noticed that the controller was sending new error on base of the OrderError, I changed the `%s` to `%w` so I'm wrapping the errors. Unfortunately still the type assertion is somehow failing.